adafruit_cursorcontrol/cursorcontrol_cursormanager.py

```diff
@@ -0,0 +1,144 @@
+    """
+    def __init__(self, cursor, buttons=None):
+        self._cursor = cursor
+        self._btns = buttons
+        self._btn_clicked = None
+        self._is_held = False
+        self._init_hardware()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.deinit()
+
+    def deinit(self):
+        """Deinitializes a CursorManager object."""
+        self._is_deinited()
+        self._pad.deinit()
+        self._cursor.deinit()
+        self._cursor = None
+
+    def _is_deinited(self):
+        """Checks if CursorManager object has been deinitd."""
+        if self._cursor is None:
+            raise ValueError("CursorManager object has been deinitialized and can no longer "
+                             "be used. Create a new CursorManager object.")
+
+    def _init_hardware(self):
+        """Initializes PyBadge or PyGamer hardware."""
+        if hasattr(board, 'BUTTON_CLOCK') and not hasattr(board, 'JOYSTICK_X'):
+            self._pad_btns = {'btn_left' : PYBADGE_BUTTON_LEFT,
+                              'btn_right' : PYBADGE_BUTTON_RIGHT,
+                              'btn_up' : PYBADGE_BUTTON_UP,
+                              'btn_down' : PYBADGE_BUTTON_DOWN,
+                              'btn_a' : PYBADGE_BUTTON_A,
+                              'btn_b' : PYBADGE_BUTTON_B}
+        elif hasattr(board, 'JOYSTICK_X'):
+            self._joystick_x = analogio.AnalogIn(board.JOYSTICK_X)
+            self._joystick_y = analogio.AnalogIn(board.JOYSTICK_Y)
+            self._pad_btns = {'btn_a' : PYBADGE_BUTTON_A,
+                              'btn_b' : PYBADGE_BUTTON_B}
+        else:
+            raise AttributeError('Board must have a D-Pad or Joystick for use with CursorManager!')
+        self._pad = GamePadShift(digitalio.DigitalInOut(board.BUTTON_CLOCK),
+                                 digitalio.DigitalInOut(board.BUTTON_OUT),
+                                 digitalio.DigitalInOut(board.BUTTON_LATCH))
+
+    def update(self):
+        """Updates the cursor object.
+
+        Returns:
+            int: Optionally returns the number of the display button clicked.
+            None: If no display buton object was clicked.
+        """
+        pressed = self._pad.get_pressed()
+        self._check_cursor_movement(pressed)
+        if self._is_held:
+            if not pressed & (self._pad_btns['btn_a'] | self._pad_btns['btn_b']):
+                self._is_held = False
+                for i, self._btn_clicked in enumerate(self._btns): # pylint: disable=unused-variable
+                    self._btn_clicked.selected = False
+        if self._btns: # user-defined buttons provided to cursormanager
+            button_num = self._check_cursor_click(pressed)
+            if self._btn_clicked is not None:
+                return button_num
+        return None
+
+    def _check_cursor_click(self, pressed):
```

Comment:
this is odd to have in the library, the cursor control shouldn't know or keep track of buttons in the UI - that's up the UI to decide! have it so the UI can get click/movement from the cursor, but objects should be individualistic, not interacting with other objects that are not under their purview

Reply:
Ok - I'll add a `is_clicked` property to handle btn presses.